metabase/metabase
# Migrate snippet collection form to formik

Epic #26178
Based on #26928 

1. Migrates snippet collection form component to `formik`
2. Breaks the modal form component into a form component and a modal component
3. Covers both components with unit tests
4. Cleans up snippet collections entity

### To Verify

1. Spin up paid Metabase environment since snippet collections are a paid feature
2. New > SQL query
3. Click the snippets icon on the right side of the query content editor (right above the "Run" button)
4. Click the "+" icon at the top right corner of the snippets sidebar and pick "Folder"
5. Ensure you can't submit the form until you fill in a name
6. Ensure you can create a folder (with/without description; with default/changed parent folder)
7. Once a folder is created, click the ellipsis icon next to edit and pick "Edit"
8. Ensure you can edit collection (one property, many properties at the same time, etc.)

### Demo

https://user-images.githubusercontent.com/17258145/205361801-0d36821e-822c-4c21-806a-cf082643194d.mp4
